### PR TITLE
Elasticsearch: Remove any type from elastic_response.test.ts

### LINKED ISSUE #53400: Elasticsearch: Remove `any` public/app/plugins/datasource/elasticsearch/specs/elastic_response.test.ts
In `public/app/plugins/datasource/elasticsearch/specs/elastic_response.test.ts` we have couple of `any` types or `as any` type assertions. The goal is to remove all of them from this test file.

### PR DESCRIPTION
**Pull request by itsgareth**
**What this PR does / why we need it**:

Remove any from `public/app/plugins/datasource/elasticsearch/specs/elastic_response.test.ts`

**Which issue(s) this PR fixes**:

Fixes #53400

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/datasource/elasticsearch/specs/elastic_response.test.ts [hunk 1]
```diff
@@ -1469,3 +1489,41 @@ describe('ElasticResponse', () => {
     });
   });
 });
+
+interface MockedElasticResponse {
```
Comment (matyax) [inline public/app/plugins/datasource/elasticsearch/specs/elastic_response.test.ts:1493]: @gwdawson I just wanted to double check these, to see if we really need them. Can't we use existing interfaces outside of this test file?

**Review by matyax — COMMENTED** [review]
Looks good! Left a question.

File: public/app/plugins/datasource/elasticsearch/specs/elastic_response.test.ts (hunk 1 above)
Comment (gwdawson) [reply public/app/plugins/datasource/elasticsearch/specs/elastic_response.test.ts:1493]: just double checked and it seems there isn't already a pre-existing interface to fit this.

Comment (matyax) [reply public/app/plugins/datasource/elasticsearch/specs/elastic_response.test.ts:1493]: Thank you for checking!

**Review by matyax — APPROVED** [review]
Let's go!

**Pull request merged**